server/src/controllers/userController.ts

```diff
@@ -0,0 +1,171 @@
+import { Request, Response } from 'express'
+import { v4 } from 'uuid'
+import { validateUserSchema } from '../utils/validation'
+import {
+  GenerateOTP,
+  GenerateToken,
+  hashPassword,
+  passwordUtils,
+} from '../utils/helpers'
+import Users, { UserAttributes, role } from '../models/users'
+import { mailOTP } from '../utils/mailFunctions'
+import Wallets, { WalletAttributes, type } from '../models/wallets'
+import Settings, { SettingsAttribute } from '../models/settings'
+
+export const registerUser = async (req: Request, res: Response) => {
+  try {
+    const { firstName, lastName, email, phone, password, confirm_password } =
+      req.body
+
+    const error = validateUserSchema.safeParse(req.body)
+    if (error.success === false) {
+      return res.status(400).send({
+        status: 'error',
+        method: req.method,
+        message: error.error.issues,
+      })
+    }
+    const passwordRegex = passwordUtils.regex
+    if (!passwordRegex.test(password)) {
+      return res.status(400).json({
+        status: 'error',
+        message: passwordUtils.error,
+      })
+    }
+    if (password !== confirm_password)
+      return res.status(400).json({
+        status: 'error',
+        message: `Password mismatch`,
+      })
```

Comment:
Frontend usually validates this, so there's no need to do this at the backend

Reply:
well, I feel it's like a second layer of protection. Just like when validate inputs using zod or joi even though they have probably been validated in the front end forms. Or is it best I remove it?

Reply:
I was thinking of an extensive folder structure for user controllers where we have a folder each for auth, verifyToken, savingsWallet, groupWallet. This could make our code more readable and avoid instances of files with 1000lines of code or more. WDYT? Because user related logics and endpoints in this app are quite broad